components/script/dom/customelementregistry.rs

```diff
@@ -95,7 +96,15 @@ impl CustomElementRegistryMethods for CustomElementRegistry {
         rooted!(in(global_scope.get_cx()) let constructor = constructor_.callback());
 
         // Step 1
-        if unsafe { !IsConstructor(constructor.get()) } {
+        // We must unwrap the constructor as all wrappers are constructable if they are callable.
+        rooted!(in(global_scope.get_cx()) let unwrapped_constructor = unsafe { UnwrapObject(constructor.get(), 1) });
```

Comment:
That's an interesting question I looked more carefully at Gecko, and they have two different hashmaps. One is keyed on the unwrapped constructor and stores local name values. The other table is keyed on local name values and stores the CustomElementDefinition, which contains the original constructor value. This means that all lookups by the constructor pointer are performed on unwrapped constructors, but the actual value that originated from script is used.